### PR TITLE
Implement vue coding guidelines

### PR DESCRIPTION
**Pull request by erikherberg**
### Changes

- Update dependencies to latest version
- Include new linting rules based upon our vue coding guidelines

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,26 +18,26 @@
     "url": "https://github.com/team23/eslint-config-team23-vue"
   },
   "dependencies": {
-    "@team23/eslint-config-team23-standard": "^8.2.0",
+    "@team23/eslint-config-team23-standard": "^8.3.0",
     "@team23/eslint-config-team23-ts": "^6.0.0",
-    "eslint-plugin-vue": "^9.17.0"
+    "eslint-plugin-vue": "^9.21.1"
   },
   "devDependencies": {
```
Comment (goester-team23) [inline package.json:25]: **nitpick**: misses vue as dev dependency. Vue is used in TestComponent.vue

File: test/TestComponent.vue [hunk 2]
```diff
@@ -1,3 +1,19 @@
+<script lang="ts">
```
Comment (goester-team23) [inline test/TestComponent.vue]: **suggestion**: use a script setup component to reflect our current standards.

File: .eslintrc.js [hunk 3]
```diff
@@ -22,14 +22,53 @@ module.exports = {
         'plugin:vue/vue3-recommended',
         'plugin:vuejs-accessibility/recommended',
     ],
+    rules: {
+        'vue/block-lang': ['error', {
```
Comment (goester-team23) [inline .eslintrc.js]: **question**: why did you place the rules here instead of the vue block in the overrides?

**Review by goester-team23 — COMMENTED** [review]
Reviewed with comments

File: test/TestComponent.vue [hunk 4]
```diff
@@ -1,3 +1,19 @@
...
+import { ref } from 'vue';
```
Comment (goester-team23) [inline test/TestComponent.vue]: **suggestion**: you may want to add more tests

File: .eslintrc.js (hunk 3 above)
Comment (erikherberg) [reply .eslintrc.js]: @goester-team23 These rules are optional and not previously declared. That is why we declare them for the first time.

File: test/TestComponent.vue (hunk 4 above)
Comment (erikherberg) [reply test/TestComponent.vue]: @goester-team23 Added some additional examples. May add more in the future.

**Review by goester-team23 — APPROVED** [review]
Nice work!

**Pull request merged**